crates/pixi_core/src/lock_file/satisfiability/mod.rs

```diff
@@ -1689,103 +1694,125 @@ pub(crate) async fn verify_package_platform_satisfiability(
+
+        async move {
+            // Build source metadata spec to request current package metadata
+            let source_metadata_spec = SourceMetadataSpec {
+                package: source_record.package_record.name.clone(),
+                backend_metadata: BuildBackendMetadataSpec {
+                    manifest_source: source_record.manifest_source.clone(),
+                    channel_config,
+                    channels: channel_urls,
+                    build_environment: BuildEnvironment {
+                        host_platform: platform,
+                        build_platform: platform,
+                        host_virtual_packages: virtual_packages.clone(),
+                        build_virtual_packages: virtual_packages,
+                    },
+                    variants: Some(variants),
+                    variant_files: None,
+                    enabled_protocols: EnabledProtocols::default(),
+                    pin_override: source_record.build_source.clone(),
+                },
             };
-            path_record.resolve(project_root)
-        };
-
-        let source_dir = source_dir.canonicalize().map_err(|e| {
-            Box::new(PlatformUnsat::FailedToCanonicalizePath(
-                source_dir.display().to_string().into(),
-                e,
-            ))
-        })?;
 
-        // Always discover the backend from the manifest directory (where pixi.toml with build config is)
-        // even if we're hashing files from a different build_source directory
-        let Some(manifest_path_record) = source_record.manifest_source.as_path() else {
-            continue;
-        };
-        let manifest_dir = manifest_path_record.resolve(project_root);
-        let manifest_dir = manifest_dir.canonicalize().map_err(|e| {
-            Box::new(PlatformUnsat::FailedToCanonicalizePath(
-                manifest_path_record.path.as_str().into(),
-                e,
-            ))
-        })?;
-
-        let discovered_backend = DiscoveredBackend::discover(
-            &manifest_dir,
-            &environment.channel_config(),
-            &EnabledProtocols::default(),
-        )
-        .map_err(PlatformUnsat::BackendDiscovery)
-        .map_err(Box::new)?;
-
-        let VariantConfig { variants, .. } = environment
-            .workspace()
-            .variants(platform)
-            .map_err(|err| Box::new(err.into()))?;
+            // Request source metadata to verify if it its still matches the locked one
+            let current_source_metadata = command_dispatcher
+                .source_metadata(source_metadata_spec)
+                .await
+                .map_err(|e| Box::new(PlatformUnsat::SourceMetadata(e)))?;
+
+            // Find the record that matches our locked package name
+            // (nichmor): In my testing, it is always and array of one element.
+            // should we just pop the first one?
+            let current_record = current_source_metadata
```

Comment:
If there are variants there can be multiple source metadata with the same package name. We have to find one that has the same metadata as the one we expect. You should technically only need to match on the `name` and `variants`